applications/EigenSolversApplication/custom_solvers/feast_eigensystem_solver.h

```diff
@@ -78,20 +77,102 @@ namespace { // helpers namespace
         KRATOS_ERROR_IF( mParam["e_r"].GetDouble() <= 0.0 ) <<
             "Invalid search radius provided" << std::endl;
 
-        KRATOS_INFO_IF( "FEASTEigensystemSolver",
-            mParam["e_min"].GetDouble() != 0.0 || mParam["e_max"].GetDouble() != 0.0 ) <<
-            "Manually defined e_min, e_max are not used for complex symmetric matrices" << std::endl;
-
-        KRATOS_INFO_IF( "FEASTEigensystemSolver",
-            mParam["search_lowest_eigenvalues"].GetBool() || mParam["search_highest_eigenvalues"].GetBool() ) <<
-            "Search for extremal eigenvalues is only available for Hermitian problems" << std::endl;
+        KRATOS_ERROR_IF( mParam["search_lowest_eigenvalues"].GetBool() || mParam["search_highest_eigenvalues"].GetBool() ) <<
+            "Search for extremal eigenvalues is only available for real symmetric problems" << std::endl;
     }
 
     template<> double SettingsHelper<double>::GetE1() {return mParam["e_min"].GetDouble();}
     template<> std::complex<double> SettingsHelper<std::complex<double>>::GetE1() {return std::complex<double>(mParam["e_mid_re"].GetDouble(), mParam["e_mid_im"].GetDouble());}
 
     template<>double SettingsHelper<double>::GetE2() {return mParam["e_max"].GetDouble();}
     template<> double SettingsHelper<std::complex<double>>::GetE2() {return mParam["e_r"].GetDouble();}
+
+    template<typename TScalar>
+    struct SortingHelper
+    {
+        SortingHelper(std::string Order) : mOrder(Order) {};
+        // void Check();
+        template<typename MatrixType, typename VectorType>
+        void SortEigenvalues(VectorType&, MatrixType&);
+        private:
+            std::string mOrder;
```

Comment:
maybe using an `enum class` would be more appropriate?

Reply:
I also thought about this, but then I need a converter from the json string to this enum. If you think this is clearer, I can do it of course.

Reply:
ok